Services/Init/classes/class.ilInitialisation.php

```diff
@@ -924,6 +932,11 @@ protected static function initGlobal($a_name, $a_class, $a_source_file = null)
 			$GLOBALS[$a_name] = $a_class;
 		}
 
+		if($DIC->offsetExists($a_name))
+		{
+			$DIC->offsetUnset($a_name);
+		}
```

Comment:
I would not recommend to include that here. This imo will just mask more general problems during initialisation and lead to problems that are hard to debug. With this one service may be inited twice (as intended for lng), but it may depend on some hard to determine order in the init sequence which service we will end with. It may even be possible that some parts of the system end up using the one service, while others use the other.

Its really sad that we cannot go all-in with the DIC because that would allow us to defer the decision which service we need to use to the point, where lng is actually used. Until that is possible I would suggest to just move this block nearer to the language initialisation, specialize it for `$a_name = "lng"` and add an appropriate comment that tells our future selfs why this code ended up there.

Reply:
@fneumann I used the english language as default to reduce the dependencies to other services (ilSetting, ilIniFile). What do you think about renaming this new method to getFallbackInstance()? 

@klees You're right, I moved the code to ilInitialisation::initLanguage().